File: solutions/HJK9810/2026-09-03/low/26939.java
import java.util.Scanner;

class Solution {
  private int value;

  private void inorder(int[] tree, int node, int N) {
    if (node > N) return;

    inorder(tree, node * 2, N);
    tree[node] = value;
    value += 1;
    inorder(tree, node  * 2 + 1, N);
  }

	public static void main(String args[]) throws Exception {
    Solution sol = new Solution();
		Scanner sc = new Scanner(System.in);
    int T = sc.nextInt();

		for(int test_case = 1; test_case <= T; test_case++) {
      int N = sc.nextInt();
      int[] tree = new int[N + 1];

      sol.value = 1;
      sol.inorder(tree, 1, N);

      System.out.printf("#%d %d %d\n", test_case, tree[1], tree[N / 2]);
		}

    sc.close();
	}
}

File: solutions/HJK9810/2026-09-03/low/26939.py
def algorithm():
    N = int(input())
    tree = [0] * (N + 1)
    value = 1

    def inorder(node):
        nonlocal value

        if node > N: return
        inorder(node * 2)
        tree[node] = value
        value += 1
        inorder(node * 2 + 1)

    inorder(1)
    return f"{tree[1]} {tree[N // 2]}"

T = int(input())
# 여러개의 테스트 케이스가 주어지므로, 각각을 처리합니다.
for test_case in range(1, T + 1):
    print(f"#{test_case} {algorithm()}")
